cli/pcluster/configure/utils.py

```diff
@@ -96,52 +107,73 @@ def prompt_iterable(message, options, default_value=None):
 
     The selected option will be the first element of the selected tuple.
     :param message: the message to show to the user
-    :param options: the list of tuple
+    :param options: a list of strings or dicts. dicts must have id as one of the keys
     :param default_value: the default value
     :return: the validated value
     """
     if not options:
         LOGGER.error("ERROR: No options found for {0}".format(message))
         sys.exit(1)
-    is_tuple = isinstance(options[0], (list, tuple))
-
+    is_dict = isinstance(options[0], dict)
     if default_value is None:
         default_value = get_default_suggestion(message, options)
 
     def input_to_parameter(user_input):
         try:
             if user_input.isdigit() and user_input != "0":
-                option_value = options[int(user_input) - 1][0] if is_tuple else options[int(user_input) - 1]
+                option_value = options[int(user_input) - 1]["id"] if is_dict else options[int(user_input) - 1]
             else:
                 option_value = user_input
         except (ValueError, IndexError):
             option_value = user_input
         return option_value
 
-    if is_tuple:
-        valid_options = [item[0] for item in options]
+    if is_dict:
+        valid_options = [item["id"] for item in options]
+        rows, header = get_rows_and_header(options)
+        return prompt(
+            message,
+            validator=lambda x: x in valid_options,
+            input_to_option=lambda x: input_to_parameter(x),
+            default_value=default_value,
+            options_to_print=rows,
+            table_header=header,
+        )
     else:
         valid_options = options
-
-    return prompt(
-        message,
-        validator=lambda x: x in valid_options,
-        input_to_option=lambda x: input_to_parameter(x),
-        default_value=default_value,
-        options_to_print=generate_printable_list(options),
-    )
+        return prompt(
+            message,
+            validator=lambda x: x in valid_options,
+            input_to_option=lambda x: input_to_parameter(x),
+            default_value=default_value,
+            options_to_print=generate_printable_list(options),
+        )
 
 
 def generate_printable_list(items):
     output = []
     for iterator, item in enumerate(items, start=1):
-        if isinstance(item, (list, tuple)):
-            output.append("{0}. {1}".format(iterator, " | ".join(item)))
-        else:
-            output.append("{0}. {1}".format(iterator, item))
+        output.append("{0}. {1}".format(iterator, item))
     return output
 
 
+def get_rows_and_header(items):
+    """
+    Return rows and header to print using tabulate.
+
+    :param items: list of dicts
+    :return: rows are all values in items. header is keys of dict
+    """
+    header = list(items[0].keys())
+    header.insert(0, "#")
```

Comment:
Minor: for very specific logic like this it would be nice to have an example the the printed result as comment in this function so we can immediately see that we are adding this `#` for pretty print

Reply:
I will add it to the comment